### ISSUE TITLE
Error stage 2

### ISSUE THREAD

**Issue by alexseys** [liip/wrench#9]
Hi, 
Wandering you could suggest what to do with this:

# GPG_TTY=$(tty)  /root/.local/bin/wrench search test
Stage2: X-GPGAuth-Authenticated should be set to true
Traceback (most recent call last):
  File "/root/.local/bin/wrench", line 11, in <module>
    sys.exit(main())
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 257, in main
    cli(obj={})
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 171, in search
    session = create_session_from_context(ctx.obj)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 72, in create_session_from_context
    session.authenticate()
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/requests_gpgauthlib/gpgauth_session.py", line 230, in authenticate
    return self.is_authenticated_with_token
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/requests_gpgauthlib/gpgauth_session.py", line 218, in is_authenticated_with_token
    raise GPGAuthStage2Exception("Login endpoint wrongly formatted")
requests_gpgauthlib.exceptions.GPGAuthStage2Exception: Login endpoint wrongly formatted

# gpg --version
gpg (GnuPG) 2.2.9
libgcrypt 1.8.3

**Comment by sephii** [comment]
Hi,

Please check the following settings in `~/.config/wrench/config.ini`:

* `server_url`: check that accessing the URL in your browser opens Passbolt
* `server_fingerprint`: check that it matches the server fingerprint

You can also try running wrench with `-vvv` and paste the output here. Also I plan to add a diagnosis command soon, that should help debugging such errors.

**Comment by alexseys** [comment]
Hi
here is output with -vvv
btw what's the point of entering http_password into config.ini file if I am asked for same on the first try of running command?
```
GPG_TTY=$(tty) /root/.local/bin/wrench -vvv search test
DEBUG:gnupg:50690: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --version
DEBUG:gnupg:stderr reader: <Thread(Thread-1, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-2, initial daemon)>
DEBUG:gnupg:chunk: b'gpg (GnuPG) 2.2.9\nlibgcrypt 1.8.3\nCopyright (C) 2018 Free Software Foundation, Inc.\nLicense GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to t'
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): passbolt01.dipr.partners.org:80
DEBUG:urllib3.connectionpool:http://passbolt_server:80 "GET /auth/verify.json?api-version=v2 HTTP/1.1" 200 None
DEBUG:gnupg:import_keys: '-----BEGIN PGP PUBLIC KEY BLOCK-----\nVersion: GnuPG v2.0.77 (GNU/Linux)\n\nmQENBFtYRb6LPAC1n5xULpZuopFd35/verylongstring'
DEBUG:gnupg:50693: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --import
DEBUG:gnupg:data copier: <Thread(Thread-3, initial daemon)>, <_io.BytesIO object at 0x7f16a3f30d58>, <_io.BufferedWriter name=5>
DEBUG:gnupg:closed output, 1756 bytes sent
DEBUG:gnupg:stderr reader: <Thread(Thread-4, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-5, initial daemon)>
DEBUG:gnupg:[GNUPG:] IMPORT_OK 0 verylongstring
DEBUG:gnupg:[GNUPG:] KEY_CONSIDERED verylongstring 0
DEBUG:gnupg:gpg: key 5B1673LKEFU3414AD900758B: passbolt_server <passbolt_server@company.com> not changed
DEBUG:gnupg:gpg: Total number processed: 1
DEBUG:gnupg:gpg:              unchanged: 1
DEBUG:gnupg:[GNUPG:] IMPORT_RES 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0
DEBUG:gnupg:import_keys result: {'gpg': <gnupg.GPG object at 0x7f16a3f52470>, 'imported': 0, 'results': [{'fingerprint': 'verylongstring', 'ok': '0', 'text': 'Not actually changed\n'}], 'fingerprints': ['verylongstring'], 'count': 1, 'no_user_id': 0, 'imported_rsa': 0, 'unchanged': 1, 'n_uids': 0, 'n_subk': 0, 'n_sigs': 0, 'n_revoc': 0, 'sec_read': 0, 'sec_imported': 0, 'sec_dups': 0, 'not_imported': 0, 'data': b'', 'stderr': '[GNUPG:] IMPORT_OK 0 verylongstring\n[GNUPG:] KEY_CONSIDERED verylongstring 0\ngpg: key 5B16734AD900758B: passbolt_server <passbolt_server@company.com> not changed\ngpg: Total number processed: 1\ngpg:              unchanged: 1\n[GNUPG:] IMPORT_RES 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n'}
DEBUG:gnupg:50697: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --list-secret-keys --fingerprint --fingerprint
DEBUG:gnupg:stderr reader: <Thread(Thread-6, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-7, initial daemon)>
DEBUG:gnupg:chunk: b'sec:-:2048:1:verylongstring:::-:::scESC:::+:::23::0:\nfpr:::::::::verylongstring:\ngrp:::::::::verylongstring:\nuid:-::::1531931516::verylongstring::user_name'
DEBUG:gnupg:line: 'sec:-:2048:1:verylongstring:::-:::scESC:::+:::23::0:'
DEBUG:gnupg:line: 'fpr:::::::::verylongstring:'
DEBUG:gnupg:line: 'grp:::::::::verylongstring:'
DEBUG:gnupg:line: 'uid:-::::1531931516::verylongstring::user_name <user_name@company.com>::::::::::0:'
DEBUG:gnupg:line: 'ssb:-:2048:1:verylongstring::::::e:::+:::23:'
DEBUG:gnupg:line: 'fpr:::::::::verylongstring:'
DEBUG:gnupg:line: 'grp:::::::::verylongstring:'
DEBUG:gnupg:50701: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --encrypt --recipient verylongstring --armor --always-trust
DEBUG:gnupg:data copier: <Thread(Thread-8, initial daemon)>, <_io.BytesIO object at 0x7f16a3f30e08>, <_io.BufferedWriter name=5>
DEBUG:gnupg:closed output, 67 bytes sent
DEBUG:gnupg:stderr reader: <Thread(Thread-9, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-10, initial daemon)>
DEBUG:gnupg:[GNUPG:] KEY_CONSIDERED verylongstring 0
DEBUG:gnupg:[GNUPG:] ENCRYPTION_COMPLIANCE_MODE 23
DEBUG:gnupg:message ignored: ENCRYPTION_COMPLIANCE_MODE, 23
DEBUG:gnupg:[GNUPG:] BEGIN_ENCRYPTION 2 9
DEBUG:gnupg:[GNUPG:] END_ENCRYPTION
DEBUG:gnupg:chunk: b'-----BEGIN PGP MESSAGE-----\n\verylongstring'
DEBUG:gnupg:encrypt result: b'-----BEGIN PGP MESSAGE-----\n\verylongstring'
DEBUG:urllib3.connectionpool:http://passbolt_server.company.com:80 "POST /auth/verify.json?api-version=v2 HTTP/1.1" 200 None
INFO:requests_gpgauthlib.gpgauth_session:server_identity_is_verified: OK
DEBUG:urllib3.connectionpool:http://passbolt_server.company.com:80 "POST /auth/login.json?api-version=v2 HTTP/1.1" 200 None
INFO:requests_gpgauthlib.gpgauth_session:Decrypting the user authentication token; password prompt expected
DEBUG:gnupg:50733: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --decrypt --always-trust
DEBUG:gnupg:data copier: <Thread(Thread-11, initial daemon)>, <_io.BytesIO object at 0x7f16a3f30d00>, <_io.BufferedWriter name=5>
DEBUG:gnupg:closed output, 1045 bytes sent
DEBUG:gnupg:stderr reader: <Thread(Thread-12, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-13, initial daemon)>
DEBUG:gnupg:[GNUPG:] ENC_TO verylongstring 1 0
DEBUG:gnupg:[GNUPG:] KEY_CONSIDERED verylongstring 0
DEBUG:gnupg:[GNUPG:] KEY_CONSIDERED verylongstring 0
DEBUG:gnupg:[GNUPG:] DECRYPTION_KEY verylongstring verylongstring -
DEBUG:gnupg:message ignored: DECRYPTION_KEY, verylongstring verylongstring -
DEBUG:gnupg:[GNUPG:] KEY_CONSIDERED verylongstring 0
DEBUG:gnupg:gpg: encrypted with 2048-bit RSA key, ID verylongstring, created 2018-07-18
DEBUG:gnupg:      "user_name <user_name@company.com>"
DEBUG:gnupg:[GNUPG:] BEGIN_DECRYPTION
DEBUG:gnupg:[GNUPG:] DECRYPTION_COMPLIANCE_MODE 23
DEBUG:gnupg:message ignored: DECRYPTION_COMPLIANCE_MODE, 23
DEBUG:gnupg:[GNUPG:] DECRYPTION_INFO 2 9
DEBUG:gnupg:gpg: uncompressing failed: Unknown compression algorithm
DEBUG:gnupg:[GNUPG:] DECRYPTION_OKAY
DEBUG:gnupg:[GNUPG:] GOODMDC
DEBUG:gnupg:[GNUPG:] END_DECRYPTION
DEBUG:gnupg:decrypt result: b''
INFO:requests_gpgauthlib.gpgauth_session:user_auth_token:
DEBUG:urllib3.connectionpool:http://passbolt_server.company.com:80 "POST /auth/login.json?api-version=v2 HTTP/1.1" 200 None
WARNING:requests_gpgauthlib.gpgauth_protocol:Stage2: X-GPGAuth-Authenticated should be set to true
Traceback (most recent call last):
  File "/root/.local/bin/wrench", line 11, in <module>
    sys.exit(main())
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 257, in main
    cli(obj={})
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 777, in __call__
    return self.main(*args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 171, in search
    session = create_session_from_context(ctx.obj)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 72, in create_session_from_context
    session.authenticate()
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/requests_gpgauthlib/gpgauth_session.py", line 230, in authenticate
    return self.is_authenticated_with_token
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/requests_gpgauthlib/gpgauth_session.py", line 218, in is_authenticated_with_token
    raise GPGAuthStage2Exception("Login endpoint wrongly formatted")
requests_gpgauthlib.exceptions.GPGAuthStage2Exception: Login endpoint wrongly formatted
```

**Comment by alexseys** [comment]
Ah, and I forgot to mention, fingerprint is correct as I tried to change it and got error:
Error: Server fingerprint "_bunch of numbers_" doesn't match expected fingerprint "_bunch of numbers_"

**Comment by sephii** [comment]
Did you redact those lines (especially the first one that mentions the decryption result)?

```
DEBUG:gnupg:decrypt result: b''
INFO:requests_gpgauthlib.gpgauth_session:user_auth_token:
DEBUG:urllib3.connectionpool:http://passbolt_server.company.com:80 "POST /auth/login.json?api-version=v2 HTTP/1.1" 200 None
```
If not, it seems there's a problem with the decryption process. Please check that you imported your Passbolt private key with `wrench import_key /path/to/keyfile.asc`. When running the `search` command you got a dialog asking for your passphrase, right?

Also please try running the following command:

`gpg --homedir ~/.config/requests_gpgauthlib/.gnupg/ --list-secret-keys`

This will show if your private key was successfully imported.

Regarding the `http_password` configuration value, that's only if your Passbolt instance is asking for HTTP authentication, for example in your web server. You can leave it empty if you don't have any additional HTTP auth.

**Comment by alexseys** [comment]
Yes I did redact, but not that line:

DEBUG:gnupg:[GNUPG:] BEGIN_DECRYPTION
DEBUG:gnupg:[GNUPG:] DECRYPTION_COMPLIANCE_MODE 23
DEBUG:gnupg:message ignored: DECRYPTION_COMPLIANCE_MODE, 23
DEBUG:gnupg:[GNUPG:] DECRYPTION_INFO 2 9
DEBUG:gnupg:gpg: uncompressing failed: Unknown compression algorithm
DEBUG:gnupg:[GNUPG:] DECRYPTION_OKAY
DEBUG:gnupg:[GNUPG:] GOODMDC
DEBUG:gnupg:[GNUPG:] END_DECRYPTION
DEBUG:gnupg:decrypt result: b''
INFO:requests_gpgauthlib.gpgauth_session:user_auth_token:
DEBUG:urllib3.connectionpool:Resetting dropped connection: server_name
DEBUG:urllib3.connectionpool:http://server_name "POST /auth/login.json?api-version=v2 HTTP/1.1" 200 None
WARNING:requests_gpgauthlib.gpgauth_protocol:Stage2: X-GPGAuth-Authenticated should be set to true

When I ran command at first try, this morning it did prompt me for a password, I used one for web interface.

Command "gpg --homedir ~/.config/requests_gpgauthlib/.gnupg/ --list-secret-keys" does show me key hash I got from web interface.

when i ran this "wrench import_key /path/to/keyfile.asc." despite a note in readme that it would ask me for some info, it did not, just imoprted successfully.
/root/.local/bin/wrench import_key /root/key-private
Key "longstring"  successfully imported.
Thank you,
Alexsey

**Comment by alexseys** [comment]
Oh and also if i remove http_password from config.ini:
/root/.local/bin/wrench search test
Traceback (most recent call last):
  File "/root/.local/bin/wrench", line 11, in <module>
    sys.exit(main())
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 257, in main
    cli(obj={})
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 171, in search
    session = create_session_from_context(ctx.obj)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 64, in create_session_from_context
    ctx_obj['config']['auth']['http_username'], ctx_obj['config']['auth']['http_password']
KeyError: 'http_password'

**Comment by sephii** [comment]
Not having the `http_password` and `http_username` config keys is not supported at the moment so you should just leave their values empty, like so: `http_password = `.

As for your issue, I think the problem is here:

```
DEBUG:gnupg:gpg: uncompressing failed: Unknown compression algorithm
DEBUG:gnupg:decrypt result: b''
INFO:requests_gpgauthlib.gpgauth_session:user_auth_token:
```

The fact that `user_auth_token` is empty shows there was an error during the decryption process. For some reason it looks like gnupg tells us the decryption was OK while in fact it wasn't.

I added some debug statements to requests_gpgauthlib to help me analyze the issue. Could you please update it using `pip install --upgrade requests_gpgauthlib` so that the version 0.1.2 is installed (you can check using `pip show requests_gpgauthlib`), try again with the `-vvv` flags, and post the output here?

**Comment by alexseys** [comment]
I have to muck "private" data per compliance 

Set http_username and http_passoword  = ''

pip show requests_gpgauthlib
Name: requests-gpgauthlib
Version: 0.1.2
Summary: requests_gpgauthlib - A requests GPGAuth authentication library
Home-page: https://github.com/liip/requests_gpgauthlib
Author: Didier 'OdyX' Raboud
Author-email: odyx@liip.ch
License: UNKNOWN
Location: /usr/local/lib/python3.6/site-packages
Requires: requests, python-gnupg
Required-by:
 

 \\\\Reimporing my key:

/root/.local/bin/wrench import_key /root/key.asc
Key 2292C88D6FC79EG1FF47014CA54FE57B10637285 successfully imported.


 /root/.local/bin/wrench -vvv search test
 
DEBUG:gnupg:61074: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --version
DEBUG:gnupg:stderr reader: <Thread(Thread-1, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-2, initial daemon)>
DEBUG:gnupg:chunk: b'gpg (GnuPG) 2.2.9\nlibgcrypt 1.8.3\nCopyright (C) 2018 Free Software Foundation, Inc.\nLicense GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>\nThis is free software: you are free to change and redistribute it.\nThere is NO WARRANTY, to t'
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): hostname.company.com:80
DEBUG:urllib3.connectionpool:http://hostname.company.com:80 "GET /auth/verify.json?api-version=v2 HTTP/1.1" 200 None
DEBUG:gnupg:import_keys: '-----BEGIN PGP PUBLIC KEY BLOCK-----\nVersion: GnuPG v2.0.22 (GNU/Linux)\n\nmQENBFtPTb8BCAC1n5xULpZuopFd35/loyTHa77J3M7JSCeTV3xWrV7PtDQRdQfl\nuPaoo/OYUMutsnDzFnyDlu72gLs9xNZ71IVKhPRodBqvYf6i7xHrV/9wjvpO6Dn+\n6wnw6ajBBcgf2w7Co4AM2LF8FzCvdMhyjnpAhu5+ozH990Bch6aZx'
DEBUG:gnupg:61077: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --import
DEBUG:gnupg:data copier: <Thread(Thread-3, initial daemon)>, <_io.BytesIO object at 0x7f816af6ad58>, <_io.BufferedWriter name=5>
DEBUG:gnupg:closed output, 1756 bytes sent
DEBUG:gnupg:stderr reader: <Thread(Thread-4, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-5, initial daemon)>
DEBUG:gnupg:[GNUPG:] IMPORT_OK 0 7E26978129185693D66YZ4805B16734AD900758B
DEBUG:gnupg:[GNUPG:] KEY_CONSIDERED 7E26978129185693D66YZ4805B16734AD900758B 0
DEBUG:gnupg:gpg: key 5B16734AD900758B: "hostname.company.com <hostname@company.com>" not changed
DEBUG:gnupg:gpg: Total number processed: 1
DEBUG:gnupg:gpg:              unchanged: 1
DEBUG:gnupg:[GNUPG:] IMPORT_RES 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0
DEBUG:gnupg:import_keys result: {'gpg': <gnupg.GPG object at 0x7f816af8c438>, 'imported': 0, 'results': [{'fingerprint': '7E26978129185693D66YZ4805B16734AD900758B', 'ok': '0', 'text': 'Not actually changed\n'}], 'fingerprints': ['7E26978129185693D66YZ4805B16734AD900758B'], 'count': 1, 'no_user_id': 0, 'imported_rsa': 0, 'unchanged': 1, 'n_uids': 0, 'n_subk': 0, 'n_sigs': 0, 'n_revoc': 0, 'sec_read': 0, 'sec_imported': 0, 'sec_dups': 0, 'not_imported': 0, 'data': b'', 'stderr': '[GNUPG:] IMPORT_OK 0 7E26978129185693D66YZ4805B16734AD900758B\n[GNUPG:] KEY_CONSIDERED 7E26978129185693D66YZ4805B16734AD900758B 0\ngpg: key 5B16734AD900758B: "hostname.company.com <hostname@company.com>" not changed\ngpg: Total number processed: 1\ngpg:              unchanged: 1\n[GNUPG:] IMPORT_RES 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n'}
DEBUG:gnupg:61081: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --list-secret-keys --fingerprint --fingerprint
DEBUG:gnupg:stderr reader: <Thread(Thread-6, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-7, initial daemon)>
DEBUG:gnupg:chunk: b'sec:-:2048:1:A54FE57B10637285:1531931516:::-:::scESC:::+:::23::0:\nfpr:::::::::2292C88D6FC79EG1FF47014CA54FE57B10637285:\ngrp:::::::::3769A0DB3DA0F67DE5BBA64AA279F75F26B50264:\nuid:-::::1531931516::4CB0677E3AAEA222D7C06D33DDCC12FDF971DD27::user name'
DEBUG:gnupg:line: 'sec:-:2048:1:A54FE57B10637285:1531931516:::-:::scESC:::+:::23::0:'
DEBUG:gnupg:line: 'fpr:::::::::2292C88D6FC79EG1FF47014CA54FE57B10637285:'
DEBUG:gnupg:line: 'grp:::::::::3769A0DB3DA0F67DE5BBA64AA279F75F26B50264:'
DEBUG:gnupg:line: 'uid:-::::1531931516::4CB0677E3AAEA222D7C06D33DDCC12FDF971DD27::user name <user@company.com>::::::::::0:'
DEBUG:gnupg:line: 'ssb:-:2048:1:A3C8AD118600D0EB:1531931516::::::e:::+:::23:'
DEBUG:gnupg:line: 'fpr:::::::::50EEB5EDC3C32E8470B61668A3C8AD118600D0EB:'
DEBUG:gnupg:line: 'grp:::::::::0D5C9C31184AB45FA73943232287E9592CC11C19:'
DEBUG:gnupg:61085: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --encrypt --recipient 7E26978129185693D66YZ4805B16734AD900758B --armor --always-trust
DEBUG:gnupg:data copier: <Thread(Thread-8, initial daemon)>, <_io.BytesIO object at 0x7f816af6ae08>, <_io.BufferedWriter name=5>
DEBUG:gnupg:closed output, 67 bytes sent
DEBUG:gnupg:stderr reader: <Thread(Thread-9, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-10, initial daemon)>
DEBUG:gnupg:[GNUPG:] KEY_CONSIDERED 7E26978129185693D66YZ4805B16734AD900758B 0
DEBUG:gnupg:[GNUPG:] ENCRYPTION_COMPLIANCE_MODE 23
DEBUG:gnupg:message ignored: ENCRYPTION_COMPLIANCE_MODE, 23
DEBUG:gnupg:[GNUPG:] BEGIN_ENCRYPTION 2 9
DEBUG:gnupg:chunk: b'-----BEGIN PGP MESSAGE-----\n\nhQEMAysrtGuH0CztAQf+LjsaST0VMC+S8LFRq2sVpCn3tet7EQdfx4dM3rGdM2LN\nPpAMbep0eJGiWSqXHWKl9h4F+GVQH8PJCfBVc0g6Wwn84rGX/HAJf/QllIDfoo2w\nC3+c60UsS0MQljOQ9QASoxQKTvnj67dF2Zw01BiYywZ69SrbZP8UDPxUsGKlbIp+\nNEPDqge/7BxbKWj9WDCHmXYIOivJESDg'
DEBUG:gnupg:[GNUPG:] END_ENCRYPTION
DEBUG:gnupg:encrypt result: b'-----BEGIN PGP MESSAGE-----\n\nhQEMAysrtGuH0CztAQf+LjsaST0VMC+S8LFRq2sVpCn3tet7EQdfx4dM3rGdM2LN\nPpAMbep0eJGiWSqXHWKl9h4F+GVQH8PJCfBVc0g6Wwn84rGX/HAJf/QllIDfoo2w\nC3+c60UsS0MQljOQ9QASoxQKTvnj67dF2Zw01BiYywZ69SrbZP8UDPxUsGKlbIp+\nNEPDqge/7BxbKWj9WDCHmXYIOivJESDgHTk+setCoQdEZOOhhMuwrKNNTOyfIdMh\nuWd67L6a65O2srIAnIDkmGkij8sMtyglyPr1dF29901ChEQNcsUqLgzQ5cg3jdQ4\n+2/1uHAb+iP9rfJ3sRjUHe6EA0qI81aQs1qVdWjr49J0AfPWYItw9ZYdbDPlHXao\nDcW6++MGtTNZhW4betab2YlqbJ2cLJU/hvqjoGyW8oZvaBvI03iqEmwk7DjzZ08I\nwtunRtvWp7YA6Hzj4bzJ2BRw6KQu0jCe6FDucWri3Nb0mCYt8gQp3plTuYOM7+vl\ngaL5b90=\n=fWpQ\n-----END PGP MESSAGE-----\n'
DEBUG:urllib3.connectionpool:http://hostname.company.com:80 "POST /auth/verify.json?api-version=v2 HTTP/1.1" 200 None
INFO:requests_gpgauthlib.gpgauth_session:server_identity_is_verified: OK
DEBUG:urllib3.connectionpool:http://hostname.company.com:80 "POST /auth/login.json?api-version=v2 HTTP/1.1" 200 None
INFO:requests_gpgauthlib.gpgauth_session:Decrypting the user authentication token; password prompt expected
DEBUG:gnupg:61117: gpg --status-fd 2 --no-tty --no-verbose --fixed-list-mode --batch --with-colons --homedir /root/.config/requests_gpgauthlib/.gnupg --decrypt --always-trust
DEBUG:gnupg:data copier: <Thread(Thread-11, initial daemon)>, <_io.BytesIO object at 0x7f816af6ae60>, <_io.BufferedWriter name=5>
DEBUG:gnupg:closed output, 1045 bytes sent
DEBUG:gnupg:stderr reader: <Thread(Thread-12, initial daemon)>
DEBUG:gnupg:stdout reader: <Thread(Thread-13, initial daemon)>
DEBUG:gnupg:[GNUPG:] ENC_TO A3C8AD118600D0EB 1 0
DEBUG:gnupg:[GNUPG:] KEY_CONSIDERED 2292C88D6FC79EG1FF47014CA54FE57B10637285 0
DEBUG:gnupg:[GNUPG:] PINENTRY_LAUNCHED 61122 curses 1.1.0 - xterm-256color -
DEBUG:gnupg:[GNUPG:] KEY_CONSIDERED 2292C88D6FC79EG1FF47014CA54FE57B10637285 0
DEBUG:gnupg:gpg: encrypted with 2048-bit RSA key, ID A3C8AD118600D0EB, created 2018-07-18
DEBUG:gnupg:      "user name <user@company.com>"
DEBUG:gnupg:gpg: public key decryption failed: Inappropriate ioctl for device
DEBUG:gnupg:[GNUPG:] ERROR pkdecrypt_failed 83918950
WARNING:gnupg:potential problem: ERROR: pkdecrypt_failed 83918950
DEBUG:gnupg:[GNUPG:] BEGIN_DECRYPTION
DEBUG:gnupg:[GNUPG:] DECRYPTION_FAILED
DEBUG:gnupg:gpg: decryption failed: No secret key
DEBUG:gnupg:[GNUPG:] END_DECRYPTION
DEBUG:gnupg:decrypt result: b''
Traceback (most recent call last):
  File "/root/.local/bin/wrench", line 11, in <module>
    sys.exit(main())
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 257, in main
    cli(obj={})
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 171, in search
    session = create_session_from_context(ctx.obj)
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/wrench/commands.py", line 72, in create_session_from_context
    session.authenticate()
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/requests_gpgauthlib/gpgauth_session.py", line 230, in authenticate
    return self.is_authenticated_with_token
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/requests_gpgauthlib/gpgauth_session.py", line 214, in is_authenticated_with_token
    user_token_result=self.user_auth_token
  File "/root/.local/venvs/passbolt-wrench/lib64/python3.6/site-packages/requests_gpgauthlib/gpgauth_session.py", line 200, in user_auth_token
    raise GPGAuthStage1Exception("Auth token decryption failed")
requests_gpgauthlib.exceptions.GPGAuthStage1Exception: Auth token decryption failed


Thank you

**Comment by sephii** [comment]
You have another issue here:

` DEBUG:gnupg:gpg: public key decryption failed: Inappropriate ioctl for device`

I think you should try again, adding `GPG_TTY=$(tty)` before the command, or export this in your profile.

**Comment by alexseys** [comment]
Ah, apologies for forgetting this.
Monday is a hard day I'll try that tomorrow and report back.
Thank you.

**Comment by alexseys** [comment]
apologies for forgetting this. We moved to another solution 

Thankyou for your support.
